File: LeetCode-Java/Remove Element.java
public class Solution {
    public int removeElement(int[] A, int elem) {
        int oldIndex = 0;
        int newIndex = -1;
        while(oldIndex < A.length)
        {
            if(A[oldIndex] != elem)
            {
                if(newIndex >= 0)
                {
                    A[newIndex] = A[oldIndex];
                    newIndex ++;
                }
            }
            else
            {
                if(newIndex < 0)
                {
                    newIndex = oldIndex;
                }
            }
            oldIndex ++;
        }
        
        return newIndex < 0 ? A.length : newIndex;
    }
}

File: Leetcode-cpp/RemoveElement.cpp
class Solution {
public:
    int removeElement(int A[], int n, int elem) {
        int realIndex = -1;
        for(int i = 0;i < n;i ++)
        {
            if(A[i] != elem)
            {
                realIndex ++;
                int temp = A[i];
                A[i] = A[realIndex];
                A[realIndex] = temp;
            }
        }
        return realIndex + 1;
    }
};
